neo4j/internal/bolt/connections.go

```diff
@@ -27,7 +27,11 @@ import (
 	"net"
 )
 
-type Neo4jErrorCallback func(context.Context, idb.Connection, *db.Neo4jError) error
+type ConnectionErrorListener interface {
+	OnNeo4jError(context.Context, idb.Connection, *db.Neo4jError) error
```

Comment:
`pool.OnNeo4jError` might call the auth manager which is fallible.

The other methods used to be fallible because of the racing locks they had. They don't any longer so this could be changed. I'm just wondering how valid it is to assume that the pool might want to start failing in those functions in the future. Not sure. What do you think?

Reply:
Those are internal, but no need to be future-proof, I'd say. If they don't return errors, let's just remove these from the signatures?

Reply:
:zap: